scipy/stats/tests/test_discrete_basic.py

```diff
@@ -279,29 +279,14 @@ def test_methods_with_lists(method, distname, args):
                         rtol=1e-15, atol=1e-15)
 
 
-@pytest.mark.parametrize(
-    'dist, x, args',
-    [  # In each of the following, at least one shape parameter is invalid
-        (stats.hypergeom, np.arange(5), (3, 3, 4)),
-        (stats.nhypergeom, np.arange(-2, 5), (5, 2, 8)),
-        (stats.bernoulli, np.arange(5), (1.5, )),
-        (stats.binom, np.arange(15), (10, 1.5)),
-        (stats.betabinom, np.arange(15), (10, -0.4, -0.5)),
-        (stats.boltzmann, np.arange(5), (-1, 4)),
-        (stats.dlaplace, np.arange(5), (-0.5, )),
-        (stats.geom, np.arange(5), (1.5, )),
-        (stats.logser, np.arange(5), (1.5, )),
-        (stats.nbinom, np.arange(15), (10, 1.5)),
-        (stats.planck, np.arange(5), (-0.5, )),
-        (stats.poisson, np.arange(5), (-0.5, )),
-        (stats.randint, np.arange(5), (5, 2)),
-        (stats.skellam, np.arange(5), (-5, -2)),
-        (stats.zipf, np.arange(5), (-2, )),
-        (stats.yulesimon, np.arange(5), (-2, ))
-    ]
-)
-def test_cdf_gh13280_regression(dist, x, args):
+@pytest.mark.parametrize('distname, args', invdistdiscrete)
+def test_cdf_gh13280_regression(distname, args):
     # Test for nan output when shape parameters are invalid
+    try:
+        dist = getattr(stats, distname)
+    except TypeError:
+        dist = distname
```

Comment:
Is this in case `distname` is a distribution object rather than its name? Why would this happen?

Reply:
> Is this in case distname is a distribution object rather than its name

Yes.

> Why would this happen?

Currently, it doesn't happen for this list. But, for `distdiscrete` and `distcont`, extra distributions are added while testing `test_discrete_basic` and `test_continuous_basic` respectively. For example, see this line:

https://github.com/scipy/scipy/blob/505a8fc6cf160562830c102919e0d6d42e0a7365/scipy/stats/tests/test_discrete_basic.py#L15

So, this case is just to ensure the test doesn't start failing if something like this is done to `invdistdiscrete` or `invdistcont` in the future.

Reply:
I'd suggest leaving this out for now, then. It can be added if/when it's needed.

Reply:
I think if you do that I can merge this. We can figure out whether we want to combine the valid and invalid lists later.

Reply:
> I'd suggest leaving this out for now, then. It can be added if/when it's needed

Done!